src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/DirectoryBasedStoreProvider.cs

```diff
@@ -0,0 +1,400 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading;
+
+namespace Internal.Cryptography.Pal
+{
+    /// <summary>
+    /// Provides an implementation of an X509Store which is backed by files in a directory.
+    /// </summary>
+    internal class DirectoryBasedStoreProvider : IStorePal
+    {
+        // {thumbprint}.1.pfx to {thumbprint}.9.pfx
+        private const int MaxSaveAttempts = 9; 
+        private const string PfxExtension = ".pfx";
+        // *.pfx ({thumbprint}.pfx or {thumbprint}.{ordinal}.pfx)
+        private const string PfxWildcard = "*" + PfxExtension;
+        // .*.pfx ({thumbprint}.{ordinal}.pfx)
+        private const string PfxOrdinalWildcard = "." + PfxWildcard;
+
+        private readonly string _storePath;
+        private List<X509Certificate2> _certificates;
+        private FileSystemWatcher _watcher;
+
+        private readonly bool _readOnly;
+
+#if DEBUG
+        static DirectoryBasedStoreProvider()
+        {
+            Debug.Assert(
+                0 == OpenFlags.ReadOnly,
+                "OpenFlags.ReadOnly is not zero, read-only detection will not work");
+        }
+#endif
+
+        internal DirectoryBasedStoreProvider(string storeName, OpenFlags openFlags)
+        {
+            if (string.IsNullOrEmpty(storeName))
+            {
+                throw new CryptographicException(SR.Arg_EmptyOrNullString);
+            }
+
+            string directoryName = GetDirectoryName(storeName);
+         
+            _storePath = Path.Combine(
+                Environment.GetEnvironmentVariable("HOME"),
```

Comment:
Yeah. Even init scripts can be ran as a different user than root (i.e. a specific service user), so having "user-scoped" certificates is still a valid case there. If we want to rely on HOME, then that's fine as long as it throws a proper exception when it's not set :smile: